test/unit/specs/core/network/injectNetworkStrategy.spec.js

```diff
@@ -57,221 +53,112 @@ const eventually = (
  */
 describe("injectNetworkStrategy", () => {
   const requestBody = JSON.stringify({ id: "myrequest" });
-  const largeRequestBody = JSON.stringify({
-    id: "mylargerequest",
-    data: new Uint8Array(10 * 1024)
-  });
-
-  let client;
-  let networkStrategy;
-  let mockServerRunning = false;
 
-  // Check to see if the mock server is running:
-  beforeAll(() => {
-    return new Promise((resolve, reject) => {
-      const timeout = setTimeout(reject, 500);
-      fetch("http://localhost:1080/status", { method: "PUT" })
-        .then(resolve)
-        .catch(reject)
-        .finally(() => clearTimeout(timeout));
-    })
-      .then(() => {
-        mockServerRunning = true;
-      })
-      .catch(() => {
-        mockServerRunning = false;
-      });
-  });
-
-  const whenMockServerIsRunningIt = (name, testFunc) => {
-    it(name, () => {
-      if (mockServerRunning) {
-        return testFunc();
-      }
-      pending("Mock server is not running");
-      return Promise.resolve();
-    });
-  };
+  const scenarios = [
+    {
+      description: "window with fetch support",
+      simulateWindowWithoutFetch: true
+    },
+    {
+      description: "window without fetch support",
+      simulateWindowWithoutFetch: false
+    }
+  ];
```

Comment:
I think we may want to rethink this. It's interesting that we're running a scenario _with_ fetch support and another scenario _without_ fetch support. If we run our unit tests on a browser that support fetch and another browser  which doesn't (Internet Explorer), then should we just _not_ simulate browsers in the test? If you think about it, when we run these tests in IE, we're running a "window with fetch support" test on a browser that doesn't and _can't_ provide fetch support.

According to our SauceLabs setup, our unit tests should be running on both kinds of browsers. Because of this, I think I would prefer _not_ simulating a window without fetch support here. Thoughts?

Reply:
I agree. I think we should just simulate the things that are available in the running environment.

Reply:
Thanks for the feedback. I'll make this update before merging.